File: LWHeroOC/Classes/CALayer+Hero.m
//
//  CALayer+Hero.m
//  Hero-ObjectiveC
//
//  Created by luca.li on 2017/1/23.
//  Copyright © 2017年 luca. All rights reserved.
//

#import "CALayer+Hero.h"

@implementation CALayer (Hero)

- (NSArray <NSDictionary *> *)animations {
    
    NSArray *keys = [self animationKeys];
    NSMutableArray *array = [NSMutableArray array];
    
    if (keys && [keys count]) {
        [keys enumerateObjectsUsingBlock:^(NSString * _Nonnull key, NSUInteger idx, BOOL * _Nonnull stop) {
            NSDictionary *dic = @{key : [[self animationForKey:key] copy]};
            [array addObject:dic];
        }];
    }
    
    return array;
}

@end

File: LWHero_swift/LWHero/Classes/CALayer+Hero.swift
//
//  CALayer+Hero.swift
//  LWHero
//
//  Created by Swift Port
//  Copyright © 2025. All rights reserved.
//

import Foundation
import QuartzCore

extension CALayer {
    public var animations: [[String: CAAnimation]] {
        var result: [[String: CAAnimation]] = []

        if let keys = animationKeys() {
            for key in keys {
                if let animation = animation(forKey: key) {
                    result.append([key: animation])
                }
            }
        }

        return result
    }
}
